composer/loggers/remote_uploader_downloader.py

```diff
@@ -18,24 +18,32 @@
 import warnings
 from multiprocessing.context import SpawnProcess
 from typing import Any, Callable, Dict, List, Optional, Set, Tuple, Type, Union
+from urllib.parse import urlparse
 
 from composer.core.state import State
 from composer.loggers.logger import Logger
 from composer.loggers.logger_destination import LoggerDestination
 from composer.utils import ObjectStore, ObjectStoreTransientError, dist, format_name_with_dist, get_file, retry
+from composer.utils.object_store import LibcloudObjectStore, S3ObjectStore, SFTPObjectStore
 
 log = logging.getLogger(__name__)
 
 __all__ = ['RemoteUploaderDownloader']
 
 
-def _build_remote_backend(object_store_cls: Type[ObjectStore], object_store_kwargs: Dict[str, Any]):
+def _build_remote_backend(remote_backend_name: str, remote_backend_kwargs: Dict[str, Any]):
+    remote_backend_name_to_cls = {'s3': S3ObjectStore, 'sftp': SFTPObjectStore, 'libcloud': LibcloudObjectStore}
+    remote_backend_cls = remote_backend_name_to_cls.get(remote_backend_name, None)
+    if remote_backend_cls is None:
+        raise ValueError(
+            f'The remote backend {remote_backend_name} is not supported. Please use one of ({list(remote_backend_name_to_cls.keys())})'
+        )
     # error: Expected no arguments to "ObjectStore" constructor
```

Comment:
I'm not really sure what this error is saying now -- is it still relevant? if its talking about the ValueError above, can we move it up a bit?

Reply:
The error was a pyright error from someone else I think. But it doesn't trigger any more, so removed.